src/ibus-chewing-engine.c

```diff
@@ -468,18 +427,18 @@ ibus_chewing_engine_restore_mode (IBusChewingEngine * self)
 	g_return_if_fail (IBUS_IS_CHEWING_ENGINE (self));
 {
 	
-        IBUS_CHEWING_LOG(DEBUG, "restore_mode() statusFlags=%x", self->_priv->statusFlags);
-        if (self->_priv->pDisplay != NULL) {
-            /* Restore Led Mode only make sense if pDisplay is available */
+        IBUS_CHEWING_LOG(DEBUG, "restore_mode() statusFlags=%x", selfp->statusFlags);
+        GdkDisplay *display = gdk_display_get_default();
+        if (display != NULL) {
+            GdkSeat *seat = gdk_display_get_default_seat(display);
+            GdkDevice *keyboard = gdk_seat_get_keyboard(seat);
```

Comment:
`gdk_seat_get_keyboard` also may return NULL

Fix those I think we can then try to run the test in weston.